components/TARGET_PSA/mbed_lib.json

```diff
@@ -1,3 +1,6 @@
 {
-    "name": "psa"
+    "name": "psa",
+    "config": {
+        "present": 1
```

Comment:
Do we need this new present or could we use `#ifdef FEATURE_PSA`?

Reply:
The macro you mentioned is always present for a given target that supports PSA regardless of whether we build with or without baremetal as it is a property of the target.

What is added in this PR is to indicate the presence of the PSA library.

TARGET_PSA = Target supports PSA
MBED_CONF_PSA_PRESENT = The PSA library is included in the build

Reply:
Why not include PSA in bare metal? I don't see how this target is very useful without PSA.

Reply:
As per the baremetal documentation,  (see PR description) PSA is not included in baremetal.

Reply:
Similar to mbedtls, we should review these and fix in the next release if possible.

Reply:
The documentation for PSA is not a specification, but just explanation of current state of things. I don't see anything preventing us having PSA in bare metal mode. Will this even work without the secure image present? Maybe it makes more sense to not build PSA targets in bare metal for now?
@maclobdell @mmahadevan108

Reply:
Bare metal mode is enabled by using `"requires": ["bare-metal"]` and this turns the build system into an "opt-in" build system i.e. any library needed by the applicationn needs to be listed in `requires`. So there is probably nothing that prevents a PSA target to run in bare metal mode but it does not work out-of-the-box. Until we fix this and have a proper "bare metal mode" then we need to guard PSA code with a flag like `MBED_CONF_PSA_PRESENT`